pennylane/plugins/expt_tensornet.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Experimental simulator plugin based on tensor network contractions
+"""
+import functools
+
+import numpy as np
+import tensornetwork as tn
+
+from pennylane.utils import _flatten
+from pennylane import Device
+from pennylane.plugins.default_qubit import I, X, Y, Z, H, CNOT, SWAP, CZ, S, T, CSWAP, \
+    Rphi, Rotx, Roty, Rotz, Rot3, CRot3, CRotx, CRoty, CRotz, CRot3, hermitian, identity, \
+    unitary
+
+# tolerance for numerical errors
+tolerance = 1e-10
+
+#========================================================
+#  device
+#========================================================
+
+
+class TensorNetwork(Device):
+    """Experimental Tensor Network simulator device for PennyLane.
+
+    Args:
+        wires (int): the number of modes to initialize the device in
+    """
+    name = 'PennyLane TensorNetwork simulator plugin'
+    short_name = 'expt.tensornet'
+    pennylane_requires = '0.7'
+    version = '0.7.0'
+    author = 'Xanadu Inc.'
+    _capabilities = {"model": "qubit", "tensor_observables": True}
+
+    _operation_map = {
+        'QubitStateVector': None,
+        'BasisState': None,
```

Comment:
Is there a technical reason that makes these non-trivial?

Reply:
Don't think so, but they are slightly different to implement than operations and measurements (which were the current goal)